app/main/api/handlers/generateAddress.ts

```diff
@@ -0,0 +1,152 @@
+import log from "app/common/logging"
+
+import { asRuntimeType, asObject } from "app/common/runtimeTypes"
+import { Wallet, ExternalFinalKey } from "app/common/runtimeTypes/storage/wallets"
+import { ChainType } from "app/common/chain/chainType"
+import { JsonSerializable } from "app/common/serializers/json"
+
+import { inject } from "app/main/utils/utils"
+import { AppStateS, WalletStorageS } from "app/main/system"
+import * as publicKey from "app/main/crypto/key/publicKey"
+import * as privateKey from "app/main/crypto/key/privateKey"
+import * as key from "app/main/crypto/key/key"
+import * as p2pkh from "app/main/crypto/address/p2pkh"
+import * as keyPath from "app/main/crypto/keyPath"
+import {
+  buildGenerateAddressError,
+  GenerateAddressResponse,
+  GenerateAddressParams,
+  GenerateAddressSuccess,
+  generateAddressErrors
+} from "app/common/runtimeTypes/ipc/address"
+
+export const MAX_EXPIRATION_DATE_DIFF = 63_072_000
```

Comment:
I would write a comment telling that this is the number of seconds in a two year time period